### DISCUSSION TITLE
How to pass a struct into slint component?

### DISCUSSION THREAD

**Discussion by evan0greenup** [slint-ui/slint#2926] [Q&A] [answered]
There is callback element inside a slint component.

But it can only be a stateless function. If I want to pass a struct with its own state and method. And let the event (click ...) to access and modify this struct and also invoke the struct's method.

What should I do?

**Comment by ogoffart** [comment, marked as answer]
You can define struct in slint with

```
struct FooBar { boo: string, baa: int }
```

Then you can use this string in a property, or a callback, like so:

```
component Comp {
    in property <FooBar> some-property;
    callback do-something(FooBar);
}
```

You can also have property or callback on `global`.

That way you can access, from native code, to struct declared in Slint by adding a callback handler.

If you want to access structs written in native code for slint, this hasn't been implemented yet, but is planed in https://github.com/slint-ui/slint/issues/1726
